### PR TITLE
Convert HTTP to Markdown

### LINKED ISSUE #7536: Convert Web/HTTP to markdown
As part of openwebdocs/project#43, I will review every HTTP pages, it is a good opportunity to move it to Markdown at the same time.

In addition, I will use this issue to track all activities needed, so, at the end, I can write a checklist for the conversion.

As an initial baseline:
Screenshot 2021-08-03 at 06 52 34

### PR DESCRIPTION
**Pull request by teoli2003**
This is the final PR that fixes #7536.

Command used:
`yarn md h2m web/http --locale en-US --mode replace` 

Then I made two commits. First the rename, then the modifications. **Don't squash merge this!**

Conversion report available at: [gist](https://gist.github.com/teoli2003/a3590aaad26b0ffea58560c1c5670b36)

Only the tables with a class `properties` are not converted. There are no other problems reported by the tool.

### CODE REVIEW

**Comment by sideshowbarker** [comment]
The first paragraph of `files/en-us/web/http/headers/content-security-policy/navigate-to/index.md` seems to have failed conversion.

```markdown
The HTTP {{HTTPHeader("Content-Security-Policy")}} (CSP)
**`navigate`\*\***`-to`** directive
restricts the URLs to which a document can initiate navigations by any means including
{{HTMLElement("form")}} (if {{CSP("form-action")}} is not specified),
{{HTMLElement("a")}}, {{DOMxRef("window.location")}}, {{DOMxRef("window.open")}}, etc.
This is an enforcement on what navigations this document initiates **not\*\*
on what this document is allowed to navigate to.
```

Can kind of see what from Looking at the HTML source:

```html
<p>The HTTP {{HTTPHeader("Content-Security-Policy")}} (CSP)
  <code><strong>navigate</strong></code><strong><code>-to</code></strong> directive
  restricts the URLs to which a document can initiate navigations by any means including
  {{HTMLElement("form")}} (if {{CSP("form-action")}} is not specified),
  {{HTMLElement("a")}}, {{DOMxRef("window.location")}}, {{DOMxRef("window.open")}}, etc.
  This is an enforcement on what navigations this document initiates <strong>not</strong>
  on what this document is allowed to navigate to.</p>
```

Anyway, I pushed a fix to the branch.

**Comment by Ryuno-Ki** [comment]
Heads up, the button currently show „squash and merge“ to me.
Is automerge (via rebase) possible to set already in this repo?

**Comment by sideshowbarker** [comment]
General comment: 144 out of the 297 — almost 50% — of the markdown files in this PR have HTML tables that failed conversion. That seems like relatively a lot — so I wonder if maybe there’s some automated way we could pre-process the sources to prevent the table conversion from failing.

**Comment by sideshowbarker** [comment]
I pushed a series of commits for some other issues I found when randomly checking a sample of files from the PR, and from running markdownlint.

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/http/basics_of_http/evolution_of_http/index.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+## HTTP/0.9 – The one-line protocol
+
+The initial version of HTTP had no version number; it has been later called 0.9 to differentiate it from the later versions. HTTP/0.9 is extremely simple: requests consist of a single line and start with the only possible method {{HTTPMethod("GET")}} followed by the path to the resource (not the URL as both the protocol, server, and port are unnecessary once connected to the server).
+
+    GET /mypage.html
+
+The response is extremely simple too: it only consisted of the file itself.
+
+    <HTML>
```
Comment (hamishwillee) [inline files/en-us/web/http/basics_of_http/evolution_of_http/index.md:35]: Even though the indentation works as a method of indicating pre/code blocks, I see using the backticks as a "safer" markup.

**Comment by hamishwillee** [comment]
> General comment: 144 out of the 297 — almost 50% — of the markdown files in this PR have HTML tables that failed conversion. That seems like relatively a lot — so I wonder if maybe there’s some automated way we could pre-process the sources to prevent the table conversion from failing.

@sideshowbarker That is probably the table at the top of every header 

<table class="properties">
  <tbody>
    <tr>
      <th scope="row">Header type</th>
      <td>{{Glossary("Response header")}}</td>
    </tr>
    <tr>
      <th scope="row">{{Glossary("Forbidden header name")}}</th>
      <td>no</td>
    </tr>
    <tr>
      <th scope="row">
        {{Glossary("CORS-safelisted response header")}}
      </th>
      <td>no</td>
    </tr>
  </tbody>
</table>

Similar to the discussion with @wbamberg on CSS, do you think we might autogenerate these from frontmatter?

**Comment by sideshowbarker** [comment]
> Similar to the discussion with @wbamberg on CSS, do you think we might autogenerate these from frontmatter?

Yeah that would certainly seem do-able without a huge level of effort. But that said, I guess it doesn’t need to block this PR — could be addressed in a follow-up

**Comment by teoli2003** [comment]
Thanks all for the thorough feedback!

So here are the next steps I plan to perform:

- [ ] Open an issue on mdn/content to get the `properties` tables fixed in the mid-term. This will keep the discussion rolling past this PR.
- [ ] Add a commit here, manually fixing the case of `<pre>` blocks translated to Markdown without backticks (that is, adding the backticks). Thanks, @hamishwillee for detecting this!
- [ ] Opening an issue on yari to get the converter updated for that case.
- [ ] Request a final round of review here, then _rebase and merge_ this PR before it rots 
(@hamishwillee: PR #7873 :wink:)

@sideshowbarker: thanks a lot for the thorough review. I didn't know about _markdownlint_; it is now installed and integrated with my editor 👍 .
@Ryuno-Ki: I don't know anything about auto-merge, but we want to do a _rebase and merge_ here.

We are close!

**Comment by Ryuno-Ki** [comment]
@teoli2003 Let me close that knowledge gap: https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request#about-auto-merge

**Comment by teoli2003** [comment]
I fixed the triple ticks for `<pre>` tags.

I think we are good to go! Please have a last look; I will open the promised discussions/follow-up later today, then _rebase and merge_.

**Review by sideshowbarker — APPROVED** [review]

**Comment by teoli2003** [comment]
_Alea jacta est._

**Pull request merged**